src/main/java/io/confluent/kafkarest/resources/v2/ConsumersResource.java

```diff
@@ -0,0 +1,310 @@
+import io.confluent.kafkarest.entities.ConsumerSubscriptionRecord;
+import io.confluent.kafkarest.entities.ConsumerSubscriptionResponse;
+import io.confluent.kafkarest.entities.ConsumerOffsetCommitRequest;
+import io.confluent.kafkarest.entities.ConsumerSeekToRequest;
+import io.confluent.kafkarest.entities.ConsumerSeekToOffsetRequest;
+import io.confluent.kafkarest.entities.ConsumerAssignmentRequest;
+import io.confluent.kafkarest.entities.ConsumerAssignmentResponse;
+import io.confluent.kafkarest.entities.ConsumerCommittedRequest;
+import io.confluent.kafkarest.entities.ConsumerCommittedResponse;
+import io.confluent.kafkarest.entities.CreateConsumerInstanceResponse;
+import io.confluent.kafkarest.entities.TopicPartitionOffset;
+import io.confluent.rest.annotations.PerformanceMetric;
+
+@Path("/consumers")
+// We include embedded formats here so you can always use these headers when interacting with
+// a consumers resource. The few cases where it isn't safe are overridden per-method
+@Produces({Versions.KAFKA_V2_JSON_BINARY_WEIGHTED_LOW, Versions.KAFKA_V2_JSON_AVRO_WEIGHTED_LOW,
+           Versions.KAFKA_V2_JSON_JSON_WEIGHTED_LOW, Versions.KAFKA_V2_JSON_WEIGHTED
+           })
+@Consumes({Versions.KAFKA_V2_JSON_BINARY, Versions.KAFKA_V2_JSON_AVRO, Versions.KAFKA_V2_JSON_JSON,
+           Versions.KAFKA_V2_JSON
+	    })
+public class ConsumersResource {
+
+  private final Context ctx;
+
+  public ConsumersResource(Context ctx) {
+    this.ctx = ctx;
+  }
+
+  @POST
+  @Valid
+  @Path("/{group}")
+  @PerformanceMetric("consumer.create+v2")
+  public CreateConsumerInstanceResponse createGroup(
+      @javax.ws.rs.core.Context UriInfo uriInfo, final @PathParam("group") String group,
+      @Valid ConsumerInstanceConfig config) {
+    if (config == null) {
+      config = new ConsumerInstanceConfig();
+    }
+    String instanceId = ctx.getKafkaConsumerManager().createConsumer(group, config);
+    String instanceBaseUri = UriUtils.absoluteUriBuilder(ctx.getConfig(), uriInfo)
+        .path("instances").path(instanceId).build().toString();
+    return new CreateConsumerInstanceResponse(instanceId, instanceBaseUri);
+  }
+
+  @DELETE
+  @Path("/{group}/instances/{instance}")
+  @PerformanceMetric("consumer.delete+v2")
+  public void deleteGroup(final @PathParam("group") String group,
+                          final @PathParam("instance") String instance) {
+    ctx.getKafkaConsumerManager().deleteConsumer(group, instance);
+  }
+
+  @POST
+  @Path("/{group}/instances/{instance}/subscription")
+  @PerformanceMetric("consumer.subscribe+v2")
+  public void subscribe( @javax.ws.rs.core.Context UriInfo uriInfo, 
+			 final @PathParam("group") String group,
+			 final @PathParam("instance") String instance,
+			 @Valid ConsumerSubscriptionRecord subscription
+			) {
+      try {
+	  ctx.getKafkaConsumerManager().subscribe(group, instance, subscription);
+      } catch(java.lang.IllegalStateException e) {
+	  throw Errors.illegalStateException(e);	  
```

Comment:
If you don't want to have to wrap every case where an `IllegalStateException` is possible with code like this, you can use an `ExceptionMapper` instead. See `ZkExceptionMapper` for an example -- it's pretty simple to setup.